FreeRTOS_ARP.c

```diff
@@ -638,17 +637,19 @@ eARPLookupResult_t eARPGetCacheEntry( uint32_t * pulIPAddress,
         ( void ) memcpy( pxMACAddress->ucBytes, xBroadcastMACAddress.ucBytes, sizeof( MACAddress_t ) );
         eReturn = eARPCacheHit;
     }
-    else if( *ipLOCAL_IP_ADDRESS_POINTER == 0UL )
+    else if( *ipLOCAL_IP_ADDRESS_POINTER == 0U )
     {
         /* The IP address has not yet been assigned, so there is nothing that
          * can be done. */
         eReturn = eCantSendPacket;
     }
     else if( *ipLOCAL_IP_ADDRESS_POINTER == *pulIPAddress )
     {
+        const void * pvCopySource = ( const void * ) ipLOCAL_MAC_ADDRESS;
+        void * pvCopyDest = ( void * ) pxMACAddress->ucBytes;
```

Comment:
Why do we need this if both data types are similar?

Reply:
Possibly not, I will check that.